spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -560,85 +719,147 @@ def compute_unit_pca_scores(recording, sorting, unit_ids=None, n_comp=3, by_elec
     elif not isinstance(unit_ids, (list, np.ndarray)):
         raise Exception("unit_ids is not a valid in valid")
 
-    # concatenate all waveforms
-    all_waveforms = np.array([])
+    if memmap:
+        if sorting.get_tmp_folder() is None:
+            tmp_folder = Path(tempfile.mkdtemp())
+            sorting.set_tmp_folder(tmp_folder)
+        else:
+            tmp_folder = sorting.get_tmp_folder()
```

Comment:
This kind of code should be in extrtactors part to simplyfy here.

Reply:
I can make tmp_folder a property: if it's None, it creates it

Reply:
COuld be an idea.

Reply:
done, see https://github.com/SpikeInterface/spikeextractors/pull/331